ang/crmUi.js

```diff
@@ -754,12 +754,20 @@
             return list;
           };
 
+          // Comparison that equivocates numbers with numeric strings
+          // ex. `isPrettyMuchEqual(12, '12'): true
+          // ex. `isPrettyMuchEqual(0, '0'): true
+          // ex. `isPrettyMuchEqual(0, ''): false
+          function isPrettyMuchEqual(v1, v2) {
+            return angular.equals('' + (typeof v1 === 'number' ? v1 : (v1 || '')), '' + (typeof v2 === 'number' ? v2 : (v2 || '')));
+          }
+
           if (ctrl.ngModel) {
             // Ensure widget is updated when model changes
             ctrl.ngModel.$render = function() {
               // Trigger change so the Select2 renders the current value,
               // but only if the value has actually changed (to avoid recursion)
-              if (!angular.equals(ctrl.ngModel.$viewValue || '', element.val())) {
+              if (!isPrettyMuchEqual(ctrl.ngModel.$viewValue, element.val())) {
```

Comment:
@artfulrobot just playing with that in my browser, your version doesn't handle the number '0':
```javascript
function isEqualish(v1, v2) {
  return (v1 || '').toString() === v2.toString();
}
isEqualish(0, '0'); // false
```

Reply:
@artfulrobot I think your code is good and I would just go ahead and merge it except that it uses a couple things from ES6 (`const` and `??`).
To-date, none of Civi's JS relies on ES6 browser-compatibility, and while you could make a strong case that it's time for that to change... I don't think sneaking it into a regression-patch is best way to flip that switch.

Reply:
@artfulrobot if your goal is clarity, then maybe reconsider my original patch that encapsulates the logic in a small but documented function. I also made that function with an eye toward reusability (it could get moved elsewhere in this file so other directives can call it). I think it might also be needed by `crmEntityref.refreshModel`.

Reply:
The reason I wanted to ensure '0' is correctly handled is more for thoroughness than a pressing use-case. APIv4-based autocompletes are *very* flexible (they even support UNIONs!) so if someone wanted to have api4 return `0` as a key they definitely could (and I can think of a couple reasons why you might want to use that, for example an autocomplete of Users might include `'0' => 'Anonymous'`) so I wouldn't want them to get bitten by a bug-waiting-to-happen if we can avoid it.